packages/cli/src/lib/updateCheck.js

```diff
@@ -0,0 +1,313 @@
+import { spawn } from 'child_process'
+import fs from 'fs'
+import os from 'os'
+import path from 'path'
+
+import boxen from 'boxen'
+import chalk from 'chalk'
+import latestVersion from 'latest-version'
+import semver from 'semver'
+
+import { getConfig } from '@redwoodjs/internal/dist/config'
+
+import { setLock, unsetLock } from './locking'
+
+import { getPaths } from './index'
+
+/**
+ * @typedef {{
+ *   localVersion: string,
+ *   remoteVersions: Map<string, string>,
+ *   checkedAt: number,
+ *   shownAt: number,
+ * }} UpdateData
+ */
+
+/**
+ * @const {number} The number of milliseconds between update checks (24 hours)
+ */
+const CHECK_PERIOD = 24 * 60 * 60_000
+
+/**
+ * @const {number} The number of milliseconds between showing a user an update notification (24 hours)
+ */
+const SHOW_PERIOD = 24 * 60 * 60_000
+
+/**
+ * @const {number} The default datetime for shownAt and checkedAt in milliseconds, corresponds to 2000-01-01T00:00:00.000Z
+ */
+export const DEFAULT_DATETIME_MS = 946684800000
```

Comment:
@Josh-Walker-GM could you have another look at all these exports? Maybe we don't need to export so many of these variables/functions

Reply:
Okay I removed one exported function that didn't need to be. The other variables/functions are used within the tests which is why they are exported.